### DISCUSSION TITLE
chrome://usb-internals can't get Device/Configuration Descriptor interactively

### DISCUSSION THREAD

**Discussion by dmalnati** [hathach/tinyusb#2465] [Q&A]
Hi, I'm starting development using tinyusb, and using the chrome://usb-internals has been helpful for exploring other USB devices to compare to my own.

Unfortunately I don't seem to be able to interactively "Get Device Descriptor" or "Get Configuration Descriptor" using that tool, though the device does actually appear to be working correctly.

Here is an image of what I see:
![image](https://github.com/hathach/tinyusb/assets/1448395/acb7fafa-209b-4449-ad1d-61c2e14528bf)

I am using a tweaked version of the cdc_msc_freertos example from the pico-sdk library of tinyusb, the usb_descriptors.c file being here ([link](https://github.com/hathach/tinyusb/blob/86c416d4c0fb38432460b3e11b08b9de76941bf5/examples/device/cdc_msc_freertos/src/usb_descriptors.c)).  Basically it's very vanilla and simple.  All I did was remove the functionality for msc.

Anyone have success using the usb-internals tool to examine the device or configuration descriptors?  Or know of another tool that does work with tinyusb?  Or know of something I might be doing incorrectly to cause this?

Again, I do know my device is being requested the device/configuration descriptors (I log the request) and windows does seem to process it correctly.

I will note that other usb devices (like a logitech mouse, which I expect is working correctly) mostly have the same issue.  I've really only found one device (Saleae Logic) which seems to work.

Thanks.

Doug

**Comment by hathach** [comment]
you can use
- lsusb on linux,macos
- https://www.uwe-sieber.de/usbtreeview_e.html on windows

**Comment by dmalnati** [comment]
Thank you!
